File: client/goAPI/admin/TGTransactionStatistics.go
package admin

/**
 * Copyright 2018-19 TIBCO Software Inc. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); You may not use this file except
 * in compliance with the License.
 * A copy of the License is included in the distribution package with this file.
 * You also may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF DirectionAny KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * File name: TGTransactionStatistics.go
 * Created on: Apr 06, 2019
 * Created by: achavan
 * SVN id: $id: $
 *
 */

// TGTransactionStatistics allows users to retrieve the transaction statistics from server
type TGTransactionStatistics interface {
	// GetAverageProcessingTime returns the average processing time for the transactions
	GetAverageProcessingTime() float64
	// GetPendingTransactionsCount returns the pending transactions count
	GetPendingTransactionsCount() int64
	// GetTransactionLoggerQueueDepth returns the queue depth of transactionLogger
	GetTransactionLoggerQueueDepth() int
	// GetTransactionProcessorsCount returns the transaction processors count
	GetTransactionProcessorsCount() int64
	// GetTransactionProcessedCount returns the processed transaction count
	GetTransactionProcessedCount() int64
	// GetTransactionSuccessfulCount returns the successful transactions count
	GetTransactionSuccessfulCount() int64
}

File: api/java/src/com/tibco/tgdb/admin/TGTransactionStatistics.java
/**
 * Copyright 2019 TIBCO Software Inc. All rights reserved.
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); You may not use this file except 
 * in compliance with the License.
 * A copy of the License is included in the distribution package with this file.
 * You also may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 *  File name :TGTransactionStatistics.java
 *  Created on: 03/28/2019
 *  Created by: nimish
 *  
 *  <p>This interface allows users to retrieve the transaction statistics from server
 *  
 *  SVN Id: $Id: TGTransactionStatistics.java 3120 2019-04-25 21:21:48Z nimish $  
 */

package com.tibco.tgdb.admin;

public interface TGTransactionStatistics {
	
	
	/**
	 * Get the transaction processors count
	 * @return transaction processors count
	 */
	long getTransactionProcessorsCount ();
	
	
	/**
	 * Get the processed transaction count 
	 * @return processed tnansaction count
	 */
	long getTransactionProcessedCount ();
	
	
	/**
	 * Get the successful transactions count
	 * @return successful transactions count
	 */
	long getTransactionSuccessfulCount ();
	
	
	/**
	 * Get the average processing time
	 * @return average processing time for the transactions
	 */
	double getAverageProcessingTime ();
	
	
	/**
	 * Get the pending transactions count
	 * @return the pending transactions count
	 */
	long getPendingTransactionsCount ();
	
	
	/**
	 * Get the queue depth of transactionLogger
	 * @return queue depth of transactionLogger
	 */
	int getTransactionLoggerQueueDepth ();

}
